lib/trino-orc/src/test/java/io/trino/orc/TestSliceDictionaryColumnWriter.java

```diff
@@ -45,17 +50,34 @@ public void testDirectConversion()
                 toIntExact(DEFAULT_MAX_COMPRESSION_BUFFER_SIZE.toBytes()),
                 () -> new StringStatisticsBuilder(toIntExact(DEFAULT_MAX_STRING_STATISTICS_LIMIT.toBytes()), new NoOpBloomFilterBuilder()));
 
-        // a single row group exceeds 2G after direct conversion
-        byte[] value = new byte[megabytes(1)];
-        ThreadLocalRandom.current().nextBytes(value);
-        Block data = RunLengthEncodedBlock.create(VARCHAR, Slices.wrappedBuffer(value), 3000);
+        int[] validCodepoints = IntStream.range(0, MAX_CODE_POINT)
+                .filter(Character::isValidCodePoint)
+                .toArray();
+        Slice randomUtf8Slice = createRandomUtf8Slice(validCodepoints, megabytes(1));
+        Block data = RunLengthEncodedBlock.create(VARCHAR, Slices.wrappedBuffer(randomUtf8Slice.byteArray()), 3000);
         writer.beginRowGroup();
         writer.writeBlock(data);
         writer.finishRowGroup();
 
         assertFalse(writer.tryConvertToDirect(megabytes(64)).isPresent());
     }
 
+    public static Slice createRandomUtf8Slice(int[] codePointSet, int lengthInBytes)
+    {
+        lengthInBytes += 64;
+        int[] codePoints = new int[lengthInBytes];
+        ThreadLocalRandom random = ThreadLocalRandom.current();
+        int totalLength = 0;
+        int i = 0;
+        while (totalLength <= lengthInBytes) {
```

Comment:
```suggestion
        while (totalLength < lengthInBytes) {
```

Reply:
this doesn't really matter as the totalLength can be increased by 1, 2, 3 or 4  but ok